pymc/distributions/distribution.py

```diff
@@ -363,7 +363,46 @@ def dist(
         return rv_out
 
 
-class SymbolicDistribution:
+class RvOpTagger:
```

Comment:
I don't think this is more reasonable than just checking if a variable shows up in `model.basic_RVs`. The sampling / logp functions are all supposed to be created within a `Model` context anyway.

Reply:
By the way, I am not saying we need to have access to the model in the new function, we can just base `model.basic_RVs` as another argument, similar to how `outputs` comes from the model anyway

Reply:
It might also simplify some of the isinstance(...) checks in the `volatile` loop

Reply:
Your argument about the rewrites convinced me. I'll take `model.basic_RVs` and pass into the compilation function.